### DISCUSSION TITLE
Supabase html form

### DISCUSSION THREAD

**Discussion by drjoeycadieux** [supabase/supabase#21792] [Questions]
Hello, 

I am trying to add supabase to a html contact form but i don't get anything in supabase dashboard.

```JS
  <script src="https://cdn.jsdelivr.net/npm/@supabase/supabase-js"></script>
    <script>
        // Replace with your actual Supabase URL and API key
        const supabaseUrl = '';
        const supabaseKey = ''; // Replace with your actual API key

        // Create Supabase client
        const { createClient } = supabaseJs;
        const supabase = createClient(supabaseUrl, supabaseKey);

        // Function to submit the form
        function submitForm() {
            var name = document.getElementById('name').value;
            var email = document.getElementById('email').value;
            var message = document.getElementById('message').value;

            // Simple validation
            if (name === '' || email === '' || message === '') {
                alert('Please fill in all fields');
                return;
            }

            // Send form data to Supabase
            supabase
                .from('contacts')
                .upsert([{ name, email, message }])
                .then(response => {
                    // Handle the response as needed
                    alert('Form submitted successfully!');
                    console.log(response);
                })
                .catch(error => {
                    // Handle errors
                    alert('Error submitting the form: ' + error.message);
                    console.error(error);
                });

            // You can reset the form after successful submission if needed.
            document.getElementById('contactForm').reset();
        }
    </script>
```
and my html code

```HTML
 <form id="contactForm">
        <label for="name">Name:</label>
        <input type="text" id="name" name="name" required>

        <label for="email">Email:</label>
        <input type="email" id="email" name="email" required>

        <label for="message">Message:</label>
        <textarea id="message" name="message" rows="4" required></textarea>

        <button type="button" onclick="submitForm()">Submit</button>
    </form>
```
My code https://github.com/drjoeycadieux/Techtackcontact

![Screenshot 2024-03-05 180332](https://github.com/supabase/supabase/assets/58401407/373fdd58-e491-465d-91b7-7d799aca4b63)

![Screenshot 2024-03-05 190154](https://github.com/supabase/supabase/assets/58401407/56ea0ee8-b573-43a2-8ea2-7ed3c2b69180)

**Comment by lukajincharadze** [comment]
It seems like you're trying to integrate Supabase with your HTML contact form to store form submissions in a Supabase table. However, there are a couple of issues in your code that might be causing it not to work properly: Here's the corrected version of your script:

<script src="https://cdn.jsdelivr.net/npm/@supabase/supabase-js"></script>
<script>
    // Replace with your actual Supabase URL and API key
    const supabaseUrl = 'YOUR_SUPABASE_URL';
    const supabaseKey = 'YOUR_SUPABASE_API_KEY'; // Replace with your actual API key

    // Create Supabase client
    const supabase = createClient(supabaseUrl, supabaseKey);

    // Function to submit the form
    function submitForm() {
        var name = document.getElementById('name').value;
        var email = document.getElementById('email').value;
        var message = document.getElementById('message').value;

        // Simple validation
        if (name === '' || email === '' || message === '') {
            alert('Please fill in all fields');
            return;
        }

        // Send form data to Supabase
        supabase
            .from('contacts')
            .upsert([{ name, email, message }])
            .then(response => {
                // Handle the response as needed
                alert('Form submitted successfully!');
                console.log(response);
            })
            .catch(error => {
                // Handle errors
                alert('Error submitting the form: ' + error.message);
                console.error(error);
            });

        // You can reset the form after successful submission if needed.
        document.getElementById('contactForm').reset();
    }
</script>
Make sure to replace 'YOUR_SUPABASE_URL' and 'YOUR_SUPABASE_API_KEY' with your actual Supabase URL and API key. Once these changes are made, your form submissions should be stored in your Supabase database.
